### DISCUSSION TITLE
Publish Functions?

### DISCUSSION THREAD

**Discussion by allen508** [DaxStudio/DaxStudio#487] [Q&A] [answered]
Hi,

I was wondering what the "Publish Functions" featured does in the Advanced menu tab. I was not able to find any documentation, but reading the code, it uploads a copy of the meta data to an API. Please can you confirm the useage?

Many thanks

**Comment by marcosqlbi** [comment, marked as answer]
It is a feature that collects the metadata of the functions available, used by DAX Guide to provide updated compatibility information about the DAX functions: https://dax.guide

**Reply by royk2022** [reply]
Im a new user of Studio Dax, and to my understanding this button  "Publish Functions" does not publish any data elsewhere, is that correct?

**Reply by dgosbell** [reply]
It does not send any user data. It only sends information about the functions that the currently connected data source supports. See https://daxstudio.org/docs/features/publish-functions/

**Comment by dgosbell** [comment]
It's a good point about the documentation. I will add a page for this when I push out the next release. 

I just did a quick count in my head and I can think of at least 16 major versions of the tabular engine we support and who knows how many service packs, minor updates and preview versions exist. These buttons provide an easy way for anyone to submit an updated function list to dax.guide but they are hidden away on the advanced tab of the options as its not a common thing that we expect people to have to do.
